src/rosdiscover/cli.py

```diff
@@ -39,6 +39,10 @@ def launch(args) -> None:
     """Simulates the architectural effects of a `roslaunch` command."""
     summary = _launch_config(args)
     output = summary.to_dict()
+
+    # Warn of any undefined nodes
+    for node_summary in summary.unresolved:
```

Comment:
Doesn't this take place after the composition stage?

Reply:
I think so - but this it to do the final report of missing components (we were already logging them along the way.)